### PR TITLE
Grow strong and big!

### PR DESCRIPTION
**Pull request by JagOfTroy**
## About The Pull Request

There was originally code written in by Skyrat that would show a visual change in your character as your 'Fitness'(now Athletics) level increased. With the changes to Athletics, you will see your size increase as you grow stronger.

## Why It's Good For The Game

Visually and flavor-wise, this just adds more immersion to the game, especially for those that like playing the stronger, muscular characters. 

## Proof Of Testing

Screenshot 2024-11-19 024727
Screenshot 2024-11-19 024843

## Changelog

:cl: Sparex
add: Added size_boost mechanics to the Athletics skill. Now you will visually grow stronger as does your power level!
/:cl:

### CODE REVIEW

**Comment by JagOfTroy** [comment]
You won't actually grow THAT large as shown in the image, it was a test to make sure the size difference would be noticeable but is beyond what legendary would give someone.

**Review by projectkepler-ru — APPROVED** [review]

File: modular_zubbers/code/datums/skills/athletics.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			5,
+			10,
+			15,
+			20,
+			30,
+			50
+		)
+	)
+	/// How much bigger your mob becomes per level (these effects don't stack together)
+	var/static/size_boost = list(0, 1/16, 1/8, 3/16, 2/8, 3/8, 4/8)
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/skills/athletics.dm:27]: No need to copypaste all the stuff you aren't changing
```suggestion
	/// How much bigger your mob becomes per level (these effects don't stack together)
	var/static/size_boost = list(0, 1/16, 1/8, 3/16, 2/8, 3/8, 4/8)
```

File: modular_zubbers/code/datums/skills/athletics.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	skill_item_path = /obj/item/clothing/gloves/boxing/golden
+
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/skills/athletics.dm:30]: ditto
```suggestion

```

File: modular_zubbers/code/datums/skills/athletics.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+/datum/skill/athletics/New()
+	. = ..()
+	levelUpMessages[SKILL_LEVEL_NOVICE] = span_nicegreen("I am just getting started on my [name] journey! I think I should be able to identify other people who are working to improve their body by sight.")
+
+/datum/skill/athletics/level_gained(datum/mind/mind, new_level, old_level, silent)
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/skills/athletics.dm:35]: ditto
```suggestion
/datum/skill/athletics/level_gained(datum/mind/mind, new_level, old_level, silent)
```

File: modular_zubbers/code/datums/skills/athletics.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    . = ..()
+    if(new_level >= SKILL_LEVEL_NOVICE && old_level < SKILL_LEVEL_NOVICE)
+        ADD_TRAIT(mind, TRAIT_EXAMINE_FITNESS, SKILL_TRAIT)
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/skills/athletics.dm:38]: ditto
```suggestion
```

File: modular_zubbers/code/datums/skills/athletics.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    var/old_gym_size = RESIZE_DEFAULT_SIZE + size_boost[old_level]
+    var/new_gym_size = RESIZE_DEFAULT_SIZE + size_boost[new_level]
+
+    if(mind.current)
+        mind.current.update_transform(new_gym_size / old_gym_size)
+
+/datum/skill/athletics/level_lost(datum/mind/mind, new_level, old_level, silent)
+    . = ..()
+    if(old_level >= SKILL_LEVEL_NOVICE && new_level < SKILL_LEVEL_NOVICE)
+        REMOVE_TRAIT(mind, TRAIT_EXAMINE_FITNESS, SKILL_TRAIT)
```
Comment (Majkl-J) [inline modular_zubbers/code/datums/skills/athletics.dm:49]: ditto
```suggestion
    . = ..()
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Also needs whitespace fixed you're using spaces instead of tabs

**Comment by JagOfTroy** [comment]
Hmmm, had no idea that VSC would still recognize those extra spaces as tabs as well. 
Will have to keep an eye out for that in the future.

**Comment by IgiariValkyr** [comment]
Would recommend some kind of toggle for it for people who don't want to scale up when they bulk up, especially for those who are large enough as is.

**Comment by JagOfTroy** [comment]
> Would recommend some kind of toggle for it for people who don't want to scale up when they bulk up, especially for those who are large enough as is.

Working on adding a toggle atm.

**Comment by MajManatee** [comment]
Does this allow already tall people just to become colossal?

**Comment by JagOfTroy** [comment]
> Does this allow already tall people just to become colossal?

It would, yes if people choose to abuse it and they most likely will because it uses RESIZE_DEFAULT_SIZE and the DEFAULT_SIZE is determined by the player in the character menu.

This is a work in progress, I'm still figuring out the @toggle and pending how to add it as as verb so people can turn it off.

There should be a way to place limitations on exactly how big people can grow which is something I'll look into as well.

Thanks for everyone's feedback.

**Comment by JagOfTroy** [comment]
Still working on this, have to learn how to make and connect front-end/back-end verb as a toggle in the menu.
Just finished exams this week so my brain is fried atm.

**Comment by Cyprex** [comment]
I don't like this, scaled sprites look bad, and you don't exactly grow an inch taller by working out.

**Comment by Cyprex** [comment]
We talked about this internally and yeah, we don't like this for the aforementioned reasons.

**Comment by JagOfTroy** [comment]
Yea, I've been absent due to class and just finished my finals so I can finally get back to looking into DM code again.

If not for the scaling, I would still like there to be something significant about maxing out fitness still. Where would the appropriate place be to discuss those ideas?

**Pull request closed without merging**